metals/src/main/scala/scala/meta/internal/metals/debug/MetalsDebugAdapter.scala

```diff
@@ -0,0 +1,54 @@
+package scala.meta.internal.metals.debug
+
+import java.util.concurrent.CompletableFuture
+import org.eclipse.lsp4j.debug.Capabilities
+import org.eclipse.lsp4j.debug.InitializeRequestArguments
+import org.eclipse.lsp4j.debug.services.IDebugProtocolClient
+import org.eclipse.lsp4j.jsonrpc.services.JsonRequest
+import scala.concurrent.ExecutionContext
+import scala.meta.internal.debug.DebugAdapter
+import scala.meta.internal.debug.JvmDebugAdapter
+import scala.meta.internal.metals.BuildTargets
+import scala.meta.internal.metals.CompilationScheduler
+import scala.meta.internal.metals.MetalsEnrichments._
+import scala.meta.internal.metals.debug.{protocol => metals}
+
+/**
+ * Metals receive requests without the full knowledge - e.g. without classpath.
+ * Here, we provide such information
+ */
+class MetalsDebugAdapter(
+    underlying: JvmDebugAdapter,
+    parametersAdapter: DebugParametersAdapter
+)(implicit ec: ExecutionContext)
+    extends DebugAdapter {
+
+  @JsonRequest
```

Comment:
I don't feel strongly about this, I just realized this doesn't seem to be necessary for DAP since all the requests are simple identifiers. For LSP the endpoints are named `textDocument/definition` so that's why we pass in the string name in `MetalsLanguageServer`. Feel free to do it only if you think it helps with consistency for our usage of `@JsonRequest`